### PR TITLE
feat: onClose function for modals

### PR DESCRIPTION
**Pull request by lkappeler**
Adds the ability to pass a onClose callback to a modal

### CODE REVIEW

File: src/components/modal/index.tsx [hunk 1]
```diff
@@ -10,15 +10,17 @@ export type ModalStyle = "white" | "dark"
 
 interface Props {
   close: () => void
+  onClose?: () => void
```
Comment (masone) [inline src/components/modal/index.tsx:13]: couln't you add code to the close function instead?

```
<Modal close={() => close(); afterClose(); }
```

Comment (masone) [reply src/components/modal/index.tsx:13]: ah, that would not cover the case where we rely on the "internal close"

Comment (lkappeler) [reply src/components/modal/index.tsx:13]: Wouldn't that break the interface of the component and we would need to update it everywhere over two projects?

**Comment by lkappeler** [comment]
@masone Will create a PR with some testis and update the story book once I finished my open taksk :)

**Pull request merged**